seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java

```diff
@@ -31,41 +38,56 @@
 import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * use in elasticsearch version >= 7.*
  */
 public class ElasticsearchRowSerializer implements SeaTunnelRowSerializer {
-    private final SeaTunnelRowType seaTunnelRowType;
     private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule())
-        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
 
+    private final SeaTunnelRowType seaTunnelRowType;
+    private final IndexInfo indexInfo;
     private final IndexSerializer indexSerializer;
-
     private final IndexTypeSerializer indexTypeSerializer;
+    private boolean hasKeys;
 
     public ElasticsearchRowSerializer(ElasticsearchVersion elasticsearchVersion, IndexInfo indexInfo, SeaTunnelRowType seaTunnelRowType) {
         this.indexTypeSerializer = IndexTypeSerializerFactory.getIndexTypeSerializer(elasticsearchVersion, indexInfo.getType());
         this.indexSerializer = IndexSerializerFactory.getIndexSerializer(indexInfo.getIndex(), seaTunnelRowType);
         this.seaTunnelRowType = seaTunnelRowType;
+        this.indexInfo = indexInfo;
+        this.hasKeys = indexInfo.getIds() != null && indexInfo.getIds().size() > 0 ? true : false;
```

Comment:
suggest use `CollectionUtils.isEmpty(indexInfo)`

Reply:
fixed it.